db/column_family.cc

```diff
@@ -1355,6 +1355,12 @@ Status ColumnFamilyData::ValidateOptions(
         "[0.0, 1.0].");
   }
 
+  if (cf_options.compaction_style == kCompactionStyleFIFO &&
+      db_options.max_open_files != -1 && cf_options.ttl > 0) {
```

Comment:
Can you explain the ttl > 0?  If ttl == 0, is it okay if max_open_files != -1 ?

Reply:
As far as we know, the `max_open_files == -1` is only needed to ensure the files' creation_time properties are read, which are used for deciding to trigger TTL compaction. When `ttl == 0`, TTL compaction is disabled.